frontends/election-manager/src/components/export_ballot_pdfs_button.test.tsx

```diff
@@ -0,0 +1,176 @@
+import { fireEvent, waitFor } from '@testing-library/react';
+import { fakeLogger, LogEventId } from '@votingworks/logging';
+import { fakeKiosk, fakeUsbDrive } from '@votingworks/test-utils';
+import { assert, usbstick } from '@votingworks/utils';
+import React from 'react';
+import { fakeFileWriter } from '../../test/helpers/fake_file_writer';
+import { renderInAppContext } from '../../test/render_in_app_context';
+import { ExportBallotPdfsButton } from './export_ballot_pdfs_button';
+
+const { UsbDriveStatus } = usbstick;
+
+jest.mock('../components/hand_marked_paper_ballot');
+
+beforeEach(() => {
+  const mockKiosk = fakeKiosk();
+  mockKiosk.getUsbDrives.mockResolvedValue([fakeUsbDrive()]);
+  const fileWriter = fakeFileWriter();
+  mockKiosk.saveAs = jest.fn().mockResolvedValue(fileWriter);
+  mockKiosk.writeFile = jest.fn().mockResolvedValue(fileWriter);
+  window.kiosk = mockKiosk;
+});
+
+afterEach(() => {
+  delete window.kiosk;
+});
+
+test('button renders properly when not clicked', () => {
+  const { queryByText, queryByTestId } = renderInAppContext(
+    <ExportBallotPdfsButton />
+  );
+
+  expect(queryByText('Export Ballot PDFs')).toHaveProperty('type', 'button');
+  expect(queryByTestId('modal')).toBeNull();
+});
+
+test('modal renders insert usb screen appropriately', async () => {
+  const usbStatuses = [
+    UsbDriveStatus.absent,
+    UsbDriveStatus.recentlyEjected,
+    UsbDriveStatus.notavailable,
+  ];
+
+  for (const usbStatus of usbStatuses) {
+    const { unmount, getByText, queryAllByAltText, queryAllByTestId } =
+      renderInAppContext(<ExportBallotPdfsButton />, {
+        usbDriveStatus: usbStatus,
+      });
+    fireEvent.click(getByText('Export Ballot PDFs'));
+    await waitFor(() => getByText('No USB Drive Detected'));
+    expect(queryAllByAltText('Insert USB Image')).toHaveLength(1);
+    expect(queryAllByTestId('modal')).toHaveLength(1);
```

Comment:
Instead of using a test ID (which is viewed as a last resort when no other queries will find what you're looking for), you can usually detect modals using a role query:

https://github.com/votingworks/vxsuite/blob/3df0503a5e13fcca49f3b4036bbb92787025393b/frontends/election-manager/src/components/smartcard_modal/smartcard_modal.test.tsx#L231